functions-framework-api/src/main/java/com/google/cloud/functions/TypedFunction.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.cloud.functions;
+
+import java.lang.reflect.Type;
+
+@FunctionalInterface
+public interface TypedFunction<RequestT, ResponseT> {
+  /**
+   * Called to service an incoming event. This interface is implemented by user code to provide the
+   * action for a given background function. If this method throws any exception (including any
+   * {@link Error}) then the HTTP response will have a 500 status code.
+   *
+   * @param arg the payload of the event, deserialized from the original JSON string.
+   * @throws Exception to produce a 500 status code in the HTTP response.
+   */
+  ResponseT apply(RequestT arg) throws Exception;
+
+  /**
+   * Override configure to set configuration options for the function.
+   *
+   * @param config mutable configuration object.
+   */
+  default void configure(Configuration config) {}
+
+  /** Configures the function contract. */
+  public static interface Configuration {
```

Comment:
is this a usual pattern - to have a static interface nested in an interface?

Reply:
I think this makes sense here since the Configuration interface is only applicable in the context of a TypedFunction. I think WireFormat is highly related to TypedFunction but could possibly stand on its own -- it could be in a separate file instead. WDYT?